rtt-target/src/rtt.rs

```diff
@@ -29,18 +29,10 @@ impl RttHeader {
         ptr::write_volatile(&mut self.max_up_channels, max_up_channels);
         ptr::write_volatile(&mut self.max_down_channels, max_down_channels);
 
-        // Copy the ID in two parts to avoid having the ID string in memory in full. The ID is
-        // copied last to make it less likely an unfinished control block is detected by the host.
-
-        ptr::copy_nonoverlapping(b"SEGG_" as *const u8, self.id.as_mut_ptr(), 5);
-
+        // The ID is copied last to make it less likely an unfinished control block is detected by the host.
+        let s = b"SEGGER RTT\0\0\0\0\0\0";
+        ptr::copy_nonoverlapping(s as *const u8, self.id.as_mut_ptr(), s.len());
```

Comment:
This write was done in two parts with the extra character overwritten on purpose to avoid ever having the magic string in memory, even as a temporary. Not sure if that's actually required or useful though, but I was being careful.

Reply:
Ahh now I get it, that's why you had the `_` there as well. If the host uses the symbol table for looking up the location of the RTT datastructure it should not be a problem, if it searches the memory blindly then it might find the magic string in the flash instead of the ram, is that it?

Reply:
So the question is how does 'probe` lookup the location (searching through any peekable address, or through the symbol table)?

Reply:
The symbol table should be and is used whenever available, but there is a fallback to scan the entire memory for the magic string. I think it's a good to keep functional, for example if you need to debug something that you don't have the exact binaries for anymore.

Reply:
Ahh got it, yea that makes sense. This fallback should only be active if running without any target file specified right. Not sure what the best thing to do in this case. One could use the string "_EGGER..." and than just write an "S" to the first location. That be similar to your impl. but does not require any pointer offset magic.

Reply:
That would work as well yes - could maybe even use `ptr::write_volatile` with an array type to ensure the write is actually volatile

Reply:
Is this change required to make it work on riscv? Or did the previous code also work?

Reply:
I started this to work to add riscv  support to `panic-rtt-target` but ended up here:)

There were some bugs in 0.4.0 of `rtt-target` that needed fixing. When doing so I made some minor changes to leverage the Rust type checker. Still I think one could make the crate less "unsafe" (I just looked a small fraction of the code).

Reply:
My question was: is *this* change (writing the `SEGGER RTT` string) required for RISCV? As far as I can tell it's doing the same as the previous code, just in a different way.